src/transaction/log_recovery.c

```diff
@@ -1811,6 +1816,12 @@ log_rv_analysis_end_checkpoint (THREAD_ENTRY * thread_p, LOG_LSA * log_lsa, LOG_
   tmp_chkpt = (LOG_REC_CHKPT *) ((char *) log_page_p->area + log_lsa->offset);
   chkpt = *tmp_chkpt;
 
+  // get max between checkpoint's and group_complete's last_ack reads
+  // *INDENT-OFF*
+  log_Gl.m_ack_stream_position = std::max ((cubstream::stream_position) log_Gl.m_ack_stream_position.load (),
```

Comment:
Summary of discussion : with current code we have potential race condition between log_complete thread and checkpoint thread, so the end checkpoint log record may contain a stream position associated with previous group instead of last group.
We agreed that log_Gl.m_ack_stream_position is not crucial for recovery process and we can leave with this.
So, we need to keep max () assignment.